pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,36 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     param (
+         [int]$retryCount = 20,
+         [int]$retryDelay = 10
+     )
+ 
+     $attempt = 0
+ 
+     while ($attempt -lt $retryCount) {
+         try {
+             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
+             
+             # Check if hns_network is null
+             if ($null -eq $hns_network) {
+                 Write-Host "Attempt $($attempt + 1) returned null. Retrying..."
```

Comment:
@wgahnagl  Does this message appear in the WICD log file?

Reply:
It appears in the log, but only if it errors. I think it's a good idea to keep because you'll only really be thinking about this file when you're debugging.

Reply:
agree